File: Lab5/Task 1/CSharp/Segment.cs
using System;

namespace CSharp
{
    public class Segment : Line
    {
        public double OxAngle { get; }

        public Segment(double[] start, double[] end) : base(start, end)
        {
            OxAngle = GetXAngle();
        }

        private double GetXAngle()
        {
            var radians =  Math.Acos(Math.Abs(this.Coordinates[1][0] - this.Coordinates[0][0]) / this.Length);
            return radians * 180 / Math.PI;
        }
    }
}

File: Lab5/Task 1/С++/С++/Segment.cpp
#include "Segment.hpp"
#include <math.h>

Segment::Segment(double* start, double* end) : Line(start, end){
    this->oXAngle = CalculateXAngle();
}

double Segment::CalculateXAngle(){
    double radians = acos(abs(this->Coordinates[1][0] - this->Coordinates[0][0])/this->Length);
    return radians * 180 / (atan(1) * 4);
}

double Segment::GetXAngle() { return this->oXAngle; }
